docs/glossary-and-resources.md

```diff
@@ -23,23 +23,44 @@
 
 ### Terms for message formatting
 
+
+- **[ABNF](https://en.wikipedia.org/wiki/Augmented_Backus%E2%80%93Naur_form)** - (Augmented Backus-Naur Form). ABNF is a method of specifying valid syntax for a specification. The MessageFormat ABNF spec defines what is and is not required for a string to be valid MessageFormat. ABNF is specified by RFC 5234 and RFC 7405; x
 - **API** - any function call(s) invoked by the user to perform message formatting.
 - **API argument format** - syntax of values passed to input of message formatting API. May also refer to structure of values represented by the syntax. May be similar or same as _authoring format_. See _message syntax_.
 - **application locale** - the locale for formatting (or formatting resources) requested by an application.
+- **attribute** - A value in a message or a function call that provides additional information that may be needed to properly resolve. Attributes are key/value pairs expressed in the form "@key=value".
 - **authoring** - writing a message by hand that adheres to some syntax.
 - **authoring format** - syntax of message formatting inputs when constructed for a program, either manually (by a developer) or programtically (ex: a WYSIWYG tool used by a translator). May also refer to structure of values represented by the syntax. May be similar or same as _API argument format_. See _message syntax_.
+- **[CLDR](https://cldr.unicode.org/)** - Unicode's Common Locale Data Repository - a set of standardized building blocks that enable software to support languages and locales around the world.
 - **data model** - a syntax-independent description of the structure of values passed to the message formatting API.
-- **implementation** - code written to make the API achieve the intended behavior of output for a given input.
+- **format to parts** - context-aware functionality that can take a formatted string (such as a monetary amount, or a date), and return the seperated constituent parts.
+- **formatter** - in MessageFormat, a function that takes a set of parameters (eg: ```code: 'USD', amount: 3.9```), and returns formatted text that can be incorporated into the output string (eg: ```"<span class="cur"><span class="cur-symbol">$</span><span class="cur-amt">3.90</span> <span class="cur-code">USD</span>"```)
```

Comment:
This one is probably best left our for now; the shape of the value returned from a formatter is something that's explicitly left out of the spec as an implementation detail.

Your suggestion, for instance, presumes that a message is being formatted to a string.

```suggestion
```